design/cassandra/loss-of-persistence.md

```diff
@@ -0,0 +1,69 @@
+In order to use a local disk as a Persistent Volume, users can leverage [the local volume static provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner). 
+To use a local Persistent Volume for a Cassandra Cluster, the user only needs to specify the correct `storageClassName`.
+
+Let's now explore what happens when a local disk fails in Kubernetes, in some different cases:
+1. **Disk exhibits errors and slow performance but hasn't failed yet:** The Local Provisioner won't do something about it. The [local volume monitor](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/10), a solution being discussed for such problems, should provide a solution once ready.
+2. **Disk fails, mount point disappears:** The Local Provisioner won't do something about it. The [local volume monitor](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/10), a solution being discussed for such problems, should provide a solution once ready. This condition is easier to detect, so we can look into extending the Local Provisioner if we need it immediately.
+3. **Node with local disk fails:** This is the most common scenario, especially in cloud environments. In a cloud environment, local storage can be lost by VM migrations, Kubernetes or security upgrades, Node failure or fencing.
+
+The 3rd scenario, Node Failure, is the one we will focus on.
+
+
+### Cassandra Support for Member Failures
+
+Cassandra was built with fault tolerance in mind, so hardware failures were expected and included into the design.
+The normal way for a human administrator to deal with a Member failure is:
+
+1. Get the failed Member's IP address.
+2. Start a new Member with the JVM flag `-Dcassandra.replace_address_first_boot=<dead_node_ip>`. When starting, Cassandra will check if the Member has its bootstrap data. If it doesn't, it means that the Member lost its data and should stream it from other Members. 
+
+:warning: This only works if the Replication Factor is high enough. If some data have RF=1 and a Member fails, there is no-one to stream that data from.
+
+## Algorithm
+
+As we established before, our focus will be on the `Node Failure` scenario.
+
+Let's see how a Node Failure would play out.
+Assume a Kubernetes environment with a Cassandra Cluster of 3 Members using local disks, each Member on a different Node.
+Node of member-1 fails, bringing the local disk down with it.
+
+1. Administrator knows that the Node is failed and deletes the corresponding Node Object from Kubernetes.
+2. Cassandra Operator: 
+  1. Detects that a Member's PersistentVolume has NodeAffinity for a Node that does not exist.
+  2. Deletes the Member's PVC and recreates it, in order to give that Member a new empty disk.
+5. The member-1 Pod starts again. The Cassandra Sidecar:
```

Comment:
what is the trigger for the member pod starting again? does the operator do that explicitly? or does k8s machinery do it automatically once it's PVC is "accessible" again?

Reply:
Once the recreated PVC binds to a new PV, the Pod is scheduled by Kubernetes.
Before the PVC is recreated, the Pod cannot be scheduled because it has a PVC bound to a PV with NodeAffinity that cannot be satisfied, so it will be stuck at Pending.